tests/karma/unit/controllers/edit-user.js

```diff
@@ -271,12 +271,93 @@ describe('EditUserCtrl controller', function() {
       chai.expect(scope.errors).not.to.have.property('password');
     });
 
+    it('must have associated place if user type is restricted user', function() {
+      UpdateUser.returns(KarmaUtils.mockPromise());
+      dbQuery.returns(KarmaUtils.mockPromise());
+      createController();
+
+      scope.editUserModel.type = 'district-manager';
+      scope.editUserModel.facility_id = undefined;
+
+      // when
+      scope.editUser();
+
+      // expect
+      chai.expect(scope.errors).to.have.property('facility_id');
+    });
+
+    it('must have associated contact if user type is restricted user', function() {
+      UpdateUser.returns(KarmaUtils.mockPromise());
+      dbQuery.returns(KarmaUtils.mockPromise());
+      createController();
+
+      scope.editUserModel.type = 'district-manager';
+      scope.editUserModel.contact_id = undefined;
+
+      // when
+      scope.editUser();
+
+      // expect
+      chai.expect(scope.errors).to.have.property('contact_id');
+    });
+
+    it('must have associated place and contact if user type is restricted user', function() {
+      UpdateUser.returns(KarmaUtils.mockPromise());
+      dbQuery.returns(KarmaUtils.mockPromise());
+      createController();
+
+      scope.editUserModel.type = 'district-manager';
+      scope.editUserModel.facility_id = undefined;
+      scope.editUserModel.contact_id = undefined;
+
+      // when
+      scope.editUser();
+
+      // expect
+      chai.expect(scope.errors).to.have.property('facility_id');
+      chai.expect(scope.errors).to.have.property('contact_id');
+    });
+
+    it('doesn\'t need associated place and contact if user type is not restricted user', function() {
+      UpdateUser.returns(KarmaUtils.mockPromise());
+      dbQuery.returns(KarmaUtils.mockPromise());
+      createController();
+
+      scope.editUserModel.type = 'some-other-type';
+
+      // when
+      scope.editUser();
+
+      // expect
+      chai.expect(scope.errors).not.to.have.property('facility_id');
+      chai.expect(scope.errors).not.to.have.property('contact_id');
+    });
+
+
+    var jQuery;
```

Comment:
Could this be stubbed/mocked/spied, and then reset, in the standard sinon way?  Also might be neater including this declaration with all others at the top of the outer `define` function.